scenes/player/animation_player.gd

```diff
@@ -2,11 +2,52 @@
 # SPDX-License-Identifier: MPL-2.0
 extends AnimationPlayer
 
+const FIGHT_ANIMATIONS: Array[StringName] = [&"blow_right", &"blow_left", &"blow_up", &"blow_down"]
+var cancellable: bool = false
+
 @onready var player: Player = owner
+@onready var player_fighting: Node2D = %PlayerFighting
 
 
 func _process(_delta: float) -> void:
+	match player.state:
+		player.State.COZY:
+			_process_walk_idle(_delta)
+		player.State.FIGHTING:
+			_process_fighting(_delta)
+
+
+func _process_walk_idle(_delta: float) -> void:
 	if player.velocity.is_zero_approx():
 		play(&"idle")
 	else:
 		play(&"walk")
+
+
+func _process_fighting(_delta: float) -> void:
+	if not player_fighting.is_fighting and current_animation not in FIGHT_ANIMATIONS:
```

Comment:
This is not new in this branch but I think the name of `is_fighting` is a bit misleading – I expected it to mean the same as `player.state == COMBAT` but actually it means "the player is actively using the attack action".